src/controllers/profileController.js

```diff
@@ -0,0 +1,240 @@
+const { Employee } = model
+const { Role } = model
+
+ 
+
+class Profiles {
+  static create (req, res) {
+    const { telephone, firstname, lastname, address, dateOfBirth, gender } = req.body
+    const  profilePic = req.file.path
+    const { employeeId } = req.params
+    const invalidFirstName = req.t('field_error_message.first_name_message.invalid_first_name')
+    const empytFirstName = req.t('field_error_message.first_name_message.empty_first_name')
+    const invalidLastName = req.t('field_error_message.last_name_message.invalid_last_name')
+    const empytLastName = req.t('field_error_message.last_name_message.empty_last_name')
+    const emptyTelephone = req.t('profile-update.telephone.empty')
+    const invalidTelephone = req.t('profile-update.telephone.invalid')
+    const createdSuccess = req.t('profile-update.create.success')  
+    const invalidDateOfBirth = req.t('profile-update.dateOfBirth.invalid')  
+    const emptyDateOfBirth = req.t('profile-update.dateOfBirth.empty')  
+
+    const getAge = (birthDateString) => {
+    const today = new Date();
+    const birthDate = new Date(birthDateString);
+    const yearsDifference = today.getFullYear() - birthDate.getFullYear();
+      if (
+          today.getMonth() < birthDate.getMonth() ||
+         (today.getMonth() === birthDate.getMonth() && today.getDate() < birthDate.getDate())
+        ) {
+            return yearsDifference - 1;
+        }
+
+            return yearsDifference;
+       };
+      
+
+    if (firstname.length === 0) {
+      return res.status(400).json({
+        message: `${empytFirstName}`
+      })
+    }
+    if (!(/^[A-Za-z]+$/).test(firstname)) {
+      return res.status(400).json({
+        message: `${invalidFirstName}`
+      })
+    }
+    if (lastname.length === 0) {
+      return res.status(400).json({
+        message: `${empytLastName}`
+      })
+    }
+    if (!(/^[A-Za-z]+$/).test(lastname)) {
+      return res.status(400).json({
+        message: `${invalidLastName}`
+      })
+    }
+    if (telephone.length === 0) {
+      return res.status(400).json({
+        message: `${emptyTelephone}`
+      })
+    }
+    if (!(/^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im.test(telephone))) {
+      return res.status(400).json({
+        message: `${invalidTelephone}`
+      })  
+    }
+    if(dateOfBirth.length === 0){
+      return res.status(400).json({
+        message: `${emptyDateOfBirth}`
+      }) 
+      }
+      if (!(/^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/).test(dateOfBirth)) {
+        return res.status(400).json({
+          message: `${invalidDateOfBirth}`
+        })
+      }
+      if((getAge(dateOfBirth) < 18 )|| (getAge(dateOfBirth) > 60)){
+        return res.status(400).json({
+          message: "Your not of legal working age"
+        }) 
+        }
+  
+    return Profile
+      .create({
+        telephone, firstname, lastname, profilePic, address, dateOfBirth, gender, employeeId
```

Comment:
Why do you re-enter the firstname, lastname yet you already have these saved in Employee's DB. Why not just require them from the Employee's DB by including them. The same applies to the employeeID, you already used it in the params. I think it should not be required again in the body. I think you should include in the body only new information and the rest since it is existing in the database you just require it since the two models already have a relation.